pkg/wireguard/agent/agent.go

```diff
@@ -397,33 +455,82 @@ func (a *Agent) UpdatePeer(nodeName, pubKeyHex string, nodeIPv4, nodeIPv6 net.IP
 		return fmt.Errorf("failed to resolve peer endpoint address: %w", err)
 	}
 
-	peer := &peerConfig{
+        peer := &peerConfig{
 		pubKey:     pubKey,
 		endpoint:   epAddr,
 		nodeIPv4:   nodeIPv4,
 		nodeIPv6:   nodeIPv6,
 		allowedIPs: allowedIPs,
 	}
 
+	a.peerByNodeName[nodeName] = peer
+        a.nodeNameByPubKey[pubKey] = nodeName
+        if nodeIPv4 != nil {
+                a.nodeNameByNodeIP[nodeIPv4.String()] = nodeName
+        }
+        if nodeIPv6 != nil {
+                a.nodeNameByNodeIP[nodeIPv6.String()] = nodeName
+        }
+
+        err = a.reassemblyMasterHubConfig()
```

Comment:
appending all allowed IP addresses from all peer nodes here. i think this is required for all non-hub spoke peers, but this needs to be called on hub node as well?

Reply:
no need on hub node, master hub node configure all spoke node as [peer] for each since wiregard (linux kernel) should configure the routing table for hub node.